src/Util/Annotation/DocBlock.php

```diff
@@ -111,7 +114,7 @@ public static function ofMethod(\ReflectionMethod $method): self
             $method->getEndLine(),
             $method->getFileName(),
             $method->getName(),
-            $method->getDeclaringClass()->getName()
+            $className ?? $method->getDeclaringClass()->getName()
```

Comment:
If fetching the specific class is the use-case, then I suggest making the new argument mandatory, and naming `$className` so that it reflects the fact that this is a specific class in the hierarchy.

Reply:
Absolutely, it's not optional.